packages/plugin-relay/src/schema-builder.ts

```diff
@@ -172,9 +173,10 @@ schemaBuilderProto.nodeInterfaceRef = function nodeInterfaceRef() {
         ids: t.arg.idList({ required: true }),
       },
       resolve: async (root, args, context, info) =>
-        (await resolveNodes(this, context, info, args.ids as string[])) as Promise<
+        this.options.relayOptions?.nodesQueryOptions?.resolve?.(args.ids as string[], context) ??
```

Comment:
I think it makes more sense to decode the ids before passing them to the default resolver

Reply:
I think maybe the correct thing here would be to define the id(s) args using t.arg.globalID(s), so they are automatically decoded, resolveNodes would need to be updated to expect the ids to be decoded, but that is an easy change.  

The custom resolver should probably just take all the normal resolver args:
If you do that, then I think the better change here would be something like resolve: `this.option.relayOptions?.nodeQueryOptions?.resolve ?? async ...(default resolver here)`.

Reply:
i have tried to implement the custom resolve like this here: 
```typescript
resolve:
  this.options.relayOptions?.nodesQueryOptions?.resolve ??
  (async (root, args, context, info) =>
    (await resolveNodes(this, context, info, args.ids as string[])) as Promise<
      ObjectParam<SchemaTypes>
    >[]),
```
but it would throw an error about the types being incorrect.
Would the following be a good alternative? 
```typescript
resolve: async (root, args, context, info) => {
  if (this.options.relayOptions?.nodesQueryOptions?.resolve) {
    return (await this.options.relayOptions?.nodesQueryOptions?.resolve(
      this,
      args,
      context,
      info,
    )) as Promise<ObjectParam<SchemaTypes>>[];
  }

  return (await resolveNodes(this, context, info, args.ids as string[])) as Promise<
    ObjectParam<SchemaTypes>
  >[];
```

Reply:
I'd rather just cast the type of the resolve function than incur additional runtime overhead to fix a type issue.   You can probably just do 

resolve:
```ts
  this.options.relayOptions?.nodesQueryOptions?.resolve as never ??
  (async (root, args, context, info) =>
    (await resolveNodes(this, context, info, args.ids as string[])) as Promise<
      ObjectParam<SchemaTypes>
    >[]),
```

It's a bit of a hack, but getting these types right internally is less important than making sure the public API is properly type-safe and ensuring that the runtime overhead is as low as possible